lib/components/home/artist_rails/__tests__/artist_card-tests.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import 'react-native'
+import React from 'react'
+import renderer from 'react-test-renderer'
+
+import ArtistCard from '../artist_card'
+
+jest.mock('../../../spinner.js', () => 'ARSpinner')
+jest.mock('../../../opaque_image_view.js', () => 'AROpaqueImageView')
+jest.mock('../../../switch_view.js', () => 'ARSwitchView')
+
+it('renders correctly', () => {
+  const artist = {
+    __id: "QXJ0aXN0Omp1YW4tZ3Jpcw==",
+    formatted_artworks_count: "14 works, 3 for sale",
+    formatted_nationality_and_birthday: "Spanish, 1887–1927",
+    href: "/artist/juan-gris",
+    id: "juan-gris",
+    image: {
+      _urlZizbi: "https://d32dm0rphc51dk.cloudfront.net/wGMxL6TvlSORJzEHZsK9JA/large.jpg"
+    },
+    name: "Juan Gris",
+  }
+  const tree = renderer.create(<ArtistCard artist={artist} />).toJSON()
+  expect(tree).toMatchSnapshot()
+})
+
+it('hides bio when metadata missing correctly', () => {
+  const artist = {
+    __id: "QXJ0aXN0Omp1YW4tZ3Jpcw==",
+    formatted_artworks_count: "14 works, 3 for sale",
+    href: "/artist/juan-gris",
+    id: "juan-gris",
+    image: {
+      _urlZizbi: "https://d32dm0rphc51dk.cloudfront.net/wGMxL6TvlSORJzEHZsK9JA/large.jpg"
+    },
+    name: "Juan Gris",
+  }
+  const tree = renderer.create(<ArtistCard artist={artist} />).toJSON()
+  expect(tree).toMatchSnapshot()
```

Comment:
Interesting, looks like [react-test-renderer](https://github.com/thejameskyle/react-test-renderer#examples) comes with an API for this sort of thing.

It'd look something like this: 
```js
 const artist = {
    ...
    }

    const tree = renderer.create(<ArtistCard artist={artist} />)
    const heading = tree.find(element => element.type === 'Text' && [some more checks]);
    expect(heading).toBeFasly()
  })
```

However realistically, this test is probably better done as a simple unit test without snapshots [like these](https://github.com/artsy/emission/blob/master/lib/containers/__tests__/home-tests.js#L20-L27)

```js
const artist = ...  
const artistCard = new ArtistCard({artist})
expect(artistCard.renderMetadata).toBeFalsy()
```

This is because your snapshot changes could break because of completely different reasons to what the test is about. If someone renames an element in the ArtistCard, your test breaks, making it feel more brittle than it is. Then it's really quick for you to C&P that tests and add ones for the other bio `<Text>` parts